examples/08_experimental/plot_localizer_new_surface_analysis_experimental.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+
+design_matrix = glm.design_matrices_[0]
+contrast_matrix = np.eye(design_matrix.shape[1])
+
+# %%
+# At first, we create basic contrasts.
+basic_contrasts = {
+    column: contrast_matrix[i]
+    for i, column in enumerate(design_matrix.columns)
+}
+
+# %%
+# Next, we add some intermediate contrasts and
+# one :term:`contrast` adding all conditions with some auditory parts.
+basic_contrasts["audio"] = (
+    basic_contrasts["audio_left_hand_button_press"]
+    + basic_contrasts["audio_right_hand_button_press"]
+    + basic_contrasts["audio_computation"]
+    + basic_contrasts["sentence_listening"]
+)
+
+# one contrast adding all conditions involving instructions reading
+basic_contrasts["visual"] = (
+    basic_contrasts["visual_left_hand_button_press"]
+    + basic_contrasts["visual_right_hand_button_press"]
+    + basic_contrasts["visual_computation"]
+    + basic_contrasts["sentence_reading"]
+)
+
+# one contrast adding all conditions involving computation
+basic_contrasts["computation"] = (
+    basic_contrasts["visual_computation"]
+    + basic_contrasts["audio_computation"]
+)
+
+# one contrast adding all conditions involving sentences
+basic_contrasts["sentences"] = (
+    basic_contrasts["sentence_listening"] + basic_contrasts["sentence_reading"]
+)
+
+# %%
+# Finally, we create a dictionary of more relevant contrasts
+#
+# * 'left - right button press': probes motor activity
+#   in left versus right button presses.
+# * 'audio - visual': probes the difference of activity between listening
+#   to some content or reading the same type of content
+#   (instructions, stories).
+# * 'computation - sentences': looks at the activity
+#   when performing a mental computation task  versus simply reading sentences.
+#
+# Of course, we could define other contrasts,
+# but we keep only 3 for simplicity.
+
+contrasts = {
+    "left - right button press": (
+        basic_contrasts["audio_left_hand_button_press"]
+        - basic_contrasts["audio_right_hand_button_press"]
+        + basic_contrasts["visual_left_hand_button_press"]
+        - basic_contrasts["visual_right_hand_button_press"]
+    ),
+    "audio - visual": basic_contrasts["audio"] - basic_contrasts["visual"],
+    "computation - sentences": (
+        basic_contrasts["computation"] - basic_contrasts["sentences"]
+    ),
+}
+
+
+# %%
+# Let's estimate the contrasts by iterating over them.
+from nilearn import plotting
+
+fsaverage = datasets.fetch_surf_fsaverage()
+
+for index, (contrast_id, contrast_val) in enumerate(contrasts.items()):
+    print(
+        f"  Contrast {index + 1:1} out of {len(contrasts)}: "
+        f"{contrast_id}, right hemisphere"
+    )
+    # compute contrast-related statistics
+    z_score = glm.compute_contrast(contrast_val, stat_type="t")
+
+    # we plot it on the surface, on the inflated fsaverage mesh,
+    # together with a suitable background to give an impression
+    # of the cortex folding.
+    plotting.plot_surf_stat_map(
```

Comment:
- further refactoring will be possible once https://github.com/nilearn/nilearn/pull/4431 is merged

Reply:
should not prevent to merge before #4431 

just making a note for future PR